src/jit/lowerxarch.cpp

```diff
@@ -1030,6 +1034,23 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
     GenTreePtr shiftBy = tree->gtOp.gtOp2;
     GenTreePtr source  = tree->gtOp.gtOp1;
 
+#ifdef _TARGET_X86_
+    // The first operand of a GT_LSH_HI and GT_RSH_LO oper is a GT_LONG so that
+    // we can have a three operand form. Increment the srcCount.
+    if (source->OperGet() == GT_LONG)
```

Comment:
>   if (source->OperGet() == GT_LONG) [](start = 1, length = 36)

I would suggest that this condition be

if (tree->OperGet() == GT_LSH_HI || tree->OperGet() == GT_RSH_LO)
{
   assert(source->OperGet() == GT_LONG);
    .....
}